### ISSUE TITLE
annotate-load-annotations: Wrong type argument: stringp, nil

### ISSUE THREAD

**Issue by velkyel** [bastibe/annotate.el#32]
hello,

I get this backtrace:

```
Debugger entered--Lisp error: (wrong-type-argument stringp nil)
  substring-no-properties(nil)
  annotate-load-annotations()
  annotate-initialize()
  annotate-mode()
  my-prog-modes-hook()
  run-hooks(change-major-mode-after-body-hook prog-mode-hook emacs-lisp-mode-hook lisp-interaction-mode-hook)
  apply(run-hooks (change-major-mode-after-body-hook prog-mode-hook emacs-lisp-mode-hook lisp-interaction-mode-hook))
  run-mode-hooks(lisp-interaction-mode-hook)
  lisp-interaction-mode()
  command-line()
  normal-top-level()
```

when I run emacs (with toggle-debug-on-error) on my .emacs config: https://github.com/velkyel/dotfiles/blob/master/.emacs

emacs version:
GNU Emacs 25.1.1 (x86_64-apple-darwin13.4.0, NS appkit-1265.21 Version 10.9.5 (Build 13F1911)) of 2016-09-17

**Comment by bastibe** [comment]
What does your `.annotations` file contain?

**Comment by velkyel** [comment]
doesn't exists.  If it's touched (empty) then error is the same.

I think problem is that my "starting" buffer _scratch_ is in prog-mode
but (buffer-file-name) returns nil so (substrings-no-properties
(buffer-file-name)) fails.

**Comment by bastibe** [comment]
Thank you for your analysis. I believe I fixed it in 0.4.7. Can you check and make sure that your issue is fixed?

**Comment by velkyel** [comment]
it works.  thank you.
